src/Sql/Sql/Database Backup/Cmdlet/AzureSqlDatabaseLongTermRetentionBackupCmdletBase.cs

```diff
@@ -21,12 +21,70 @@
 using Microsoft.Azure.Commands.Sql.Database.Model;
 using Microsoft.Azure.Commands.Sql.Database.Services;
 using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using System.Linq;
+using System.Globalization;
+using Microsoft.Azure.Commands.Common.Exceptions;
+using System;
 
 namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
 {
-    public abstract class AzureSqlDatabaseLongTermRetentionBackupCmdletBase :
-        AzureSqlCmdletBase<IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel>, AzureSqlDatabaseBackupAdapter>
+    public abstract class AzureSqlDatabaseLongTermRetentionBackupCmdletBase<T> :
+        AzureSqlCmdletBase<IEnumerable<T>, AzureSqlDatabaseBackupAdapter>
     {
+        /// <summary>
+        /// The expected number of segments in a long term retention backup resource id.
+        /// </summary>
+        private const int LongTermRetentionBackupResourceIdSegmentsLength = 12;
+
+        /// <summary>
+        /// The expected number of segments in a long term retention backup resource id.
+        /// </summary>
+        private const int LongTermRetentionBackupResourceIdWithResourceGroupSegmentsLength = 14;
+
+        /// <summary>
+        /// Parse the longTermRetentionBackup resource Id
+        /// </summary>
+        /// <param name="resourceId"></param>
+        protected Dictionary<string, string> ParseLongTermRetentionBackupResourceId(string resourceId)
+        {
+            Dictionary<string, string> resourceSegments = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
+            string[] tokens = resourceId.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (tokens.Length != LongTermRetentionBackupResourceIdSegmentsLength
+                && tokens.Length != LongTermRetentionBackupResourceIdWithResourceGroupSegmentsLength)
+            {
+                throw new Exception($"Invalid ResourceId. resourceID: {resourceId}, tokens.Length {tokens.Length}");
+            }
+
+            int i = 0;
+            string type;
+            string name;
+            while (i < tokens.Length)
+            {
+                type = tokens[i++];
+                name = tokens[i++];
+                resourceSegments[type] = name;
+            }
+
+            try
+            {
+                resourceSegments.ContainsKey("locations");
+                resourceSegments.ContainsKey("longTermRetentionServers");
+                resourceSegments.ContainsKey("longTermRetentionDatabases");
+                resourceSegments.ContainsKey("longTermRetentionBackups");
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new AzPSArgumentException(
+                    "Invalid format of the resource identifier. A long term retention backup resource Id should follow one of the following formats: \n" +
```

Comment:
Please put all exception msg into src/Sql/Sql/Properties/Resources.resx. And refer it as `Properties.Resources.{ErrorMsg}`